src/psyclone/domain/lfric/algorithm/alg_gen.py

```diff
@@ -0,0 +1,410 @@
+                                 than gh_quadrature_xyoz.
+    '''
+    reader = FortranReader()
+    table = prog.symbol_table
+    try:
+        qr_rule_sym = table.lookup("quadrature_rule")
+    except KeyError:
+        qr_gaussian_mod = table.new_symbol(
+            "quadrature_rule_gaussian_mod", symbol_type=ContainerSymbol)
+        qr_gaussian_type = table.new_symbol(
+            "quadrature_rule_gaussian_type", symbol_type=DataTypeSymbol,
+            datatype=DeferredType(),
+            interface=ImportInterface(qr_gaussian_mod))
+        qr_rule_sym = table.new_symbol("quadrature_rule",
+                                       symbol_type=DataSymbol,
+                                       datatype=qr_gaussian_type)
+
+    if shape == "gh_quadrature_xyoz":
+        order = table.lookup_with_tag("element_order")
+        expr = reader.psyir_from_expression(
+            f"quadrature_xyoz_type({order.name}+3, {qr_rule_sym.name})", table)
+        prog.addchild(Assignment.create(Reference(qr_sym), expr))
+
+    else:
+        raise NotImplementedError(f"Initialisation for quadrature of type "
+                                  f"'{shape}' is not yet implemented.")
+
+
+def construct_kernel_args(prog, kern):
+    '''
+    Extends the supplied routine with all the declarations and initialisation
+    required for the arguments of the supplied kernel.
+
+    :param prog: the routine to which to add the declarations etc.
+    :type prog: :py:class:`psyclone.psyir.nodes.Routine`
+    :param kern: the kernel for which we are to create arguments.
+    :type kern: :py:class:`psyclone.dynamo0p3.DynKern`
+
+    :returns: object capturing all of the kernel arguments.
+    :rtype: :py:class:`psyclone.domain.lfric.KernCallInvokeArgList`
+
+    '''
+    const = LFRicConstants()
+    # Construct a list of the names of the function spaces that the field
+    # argument(s) are on. We use LFRicConstants.specific_function_space() to
+    # ensure that any 'wildcard' names in the meta-data are converted to
+    # an appropriate, specific function space.
+    function_spaces = []
+    for fspace in kern.arguments.unique_fss:
+        name = fspace.orig_name.lower()
+        function_spaces.append(const.specific_function_space(name))
+    _create_function_spaces(prog, set(function_spaces))
+
+    # Construct the argument list and add suitable symbols to the table.
+    kern_args = KernCallInvokeArgList(kern, prog.symbol_table)
+    kern_args.generate()
+
+    # Initialise field datastructures using the symbols added to the table
+    # when setting up the kernel arguments and the information on their
+    # respective function spaces extracted from the kernel metadata.
+    for sym, space in kern_args.fields:
+        initialise_field(prog, sym, space)
+
+    for qr_sym, shape in kern_args.quadrature_objects:
+        initialise_quadrature(prog, qr_sym, shape)
+
+    return kern_args
+
+
+def generate(kernel_path):
```

Comment:
I am now getting 'too many locals' from pylint here.

Reply:
Added a TODO and an ignore.